metricbeat/module/redis/key/_meta/fields.yml

```diff
@@ -0,0 +1,30 @@
+- name: key
+  type: group
+  description: >
+    `key` contains information about keys.
+  release: beta
+  fields:
+    - name: name
+      type: keyword
+      description: >
+        Key name.
+
+    - name: type
+      type: keyword
+      description: >
+        Key type as shown by `TYPE` command.
+
+    - name: length
+      type: long
+      description: >
+        Length of the key (Number of elements for lists, length for strings, cardinality for sets).
+
+    - name: value
+      type: text
```

Comment:
Do you expect users to do aggregations on this field?

Reply:
Good thing you asked :slightly_smiling_face: I am undecided about this field, the `string` type in redis is actually a generic one that can be used for all kind of things, from text strings or binary blobs to numeric atomic counters, so I am not sure about what users could do with this :thinking: 

The possible options I see we can do here:
1. It is a string, let's index it (the option implemented now, but not convinced about this).
2. Store it both as `value.text` and `value.keyword`.
3. If it parses as a number, store it in `value.number` as `double`, if not:
    1. Store is in `value.text` (and `value.keyword`?).
    2. Don't store it.
4. Let the users choose via an extra config option if they want to keep the value and/or how (I thought about adding an option to enable/disable value store, but I decided that it can be removed with processors).
5. Don't store the value at least on this PR and wait to see if this is requested.

Thinking now, I'd go for 3.2 (to support counters collection and having it as an structured field) or 5, so in both cases we can decide in the future what to do if we receive more feedback about more uses cases.

Reply:
The more I think about this, the more I think that it is actually a bad idea to store the value of strings by default, I'll go for 3.2 or 5, opinions?

Reply:
I am removing this value at the end, I don't think this is important for this metricset and we can rethink about this in the future when we feel more sure about the use cases.

Reply:
++ 

Let's wait and see if it gets requested and then we can ask questions.